src/audio/module_adapter/module/generic.c

```diff
@@ -282,6 +293,21 @@ int module_process_legacy(struct processing_module *mod,
 	return 0;
 }
 
+bool module_is_ready_to_process(struct processing_module *mod,
+				struct sof_source __sparse_cache **sources,
+				int num_of_sources,
+				struct sof_sink __sparse_cache **sinks,
+				int num_of_sinks)
+{
+	struct module_data *md = &mod->priv;
+
+	/* LL module has to be always ready for processing */
+	if (mod->dev->ipc_config.proc_domain == COMP_PROCESSING_DOMAIN_LL)
+		return true;
+
+	return md->ops->is_ready_to_process(mod, sources, num_of_sources, sinks, num_of_sinks);
+}
+
```

Comment:
I do share a bit of concern @btian1 raised that is the overhead of this function worth it. In many media frameworks, the condition "is ready to process" is coupled to the process() function. 

But I guess if this only is used by DP, and you have sensible usage DP (i.e. you actually save more by not scheduling the processs if it's not ready), then this makes more sense.

Reply:
in DP there's a need to remove it from process and put in separate function.
DP module should not be scheduled till it is ready. It usually is **not** every LL cycle. It's more "on demand" - if the module is ready it will get CPU

Checking it in process() would require schedule DP process, context switch to DP thread, calling process() and having result "nothing to process" 
is much more CPU consuming that this call performed in last LL task context.

I double checked reference design and will do a modification to be consistent: a default implementation of the method:
_return true when there is at least IBS on all inputs and at least OBS free space at sinks_
It may make both implementation easier and sometimes a bit better performance